code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1071,3 +1071,23 @@
 		connected_ai.connected_robots |= src
 		notify_ai(ROBOT_NOTIFICATION_NEW_UNIT)
 		sync()
+
+//For picking up small animals
+/mob/living/silicon/robot/MouseDrop(atom/over_object)

```

Comment:
Please refactor it to use this proc:
```
// recieve a mousedrop
/atom/proc/MouseDrop_T(atom/dropping, mob/user)
```

Where the source is the recharging station.